lsp-ui-flycheck.el

```diff
@@ -215,15 +216,17 @@ See https://github.com/emacs-lsp/lsp-mode."
     (flycheck-add-mode 'lsp-ui mode)))
 
 (defun lsp-ui-flycheck--report nil
+  "This callback is invoked when new diagnostics are received from the language server.  Invoke flycheck-buffer to update the display of errors if flycheck-mode is on and we are live reporting or we are in save-mode and the buffer is not modified."
   (and flycheck-mode
-       lsp-ui-flycheck-live-reporting
+       (or lsp-ui-flycheck-live-reporting
+	   (and lsp-ui-flycheck--save-mode (not (buffer-modified-p))))
        (flycheck-buffer)))
 
 ;; FIXME: Provide a way to disable lsp-ui-flycheck
 (defun lsp-ui-flycheck-enable (_)
   "Enable flycheck integration for the current buffer."
-  (when lsp-ui-flycheck-live-reporting
-    (setq-local flycheck-check-syntax-automatically nil))
+  (setq-local lsp-ui-flycheck--save-mode (memq 'save flycheck-check-syntax-automatically))
+  (setq-local flycheck-check-syntax-automatically nil)
```

Comment:
do we need that when live reporting is nil? The effect ATM is that if you call 2 times lsp in the same buffer the on-save action stops to work. If it is mandatory, I suggest changing the previous line to

`(setq-local lsp-ui-flycheck--save-mode (or (memq 'save flycheck-check-syntax-automatically) lsp-ui-flycheck--save-mode))